cmd/snap/cmd_get.go

```diff
@@ -77,6 +82,63 @@ func init() {
 		})
 }
 
+type ConfigValue struct {
+	Path  string
+	Value interface{}
+}
+
+type byConfigPath []ConfigValue
+
+func (s byConfigPath) Len() int      { return len(s) }
+func (s byConfigPath) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s byConfigPath) Less(i, j int) bool {
+	other := s[j].Path
+	for k, c := range s[i].Path {
```

Comment:
isn't splitting paths on the first '.' faster than this?

Reply:
Perhaps, probably... but should we worry about performance here?
Splitting on first '.' is not enough though - I think what you're suggesting would require splitting by '.' and then comparing by 1st path components, then by 2nd and so on. There may be multiple components separated by ".", e.g.
```
qemu:ubuntu-16.04-64 .../tests/main/snap-get# snap get snapctl-hooks -l bar.a
Key       Value
bar.a.aa  1
bar.a.ab  2
```
(example environment for the spread test from this branch).

Reply:
I'm not only worried about performance but also readability